sky/packages/sky/lib/widgets/switch.dart

```diff
@@ -72,13 +75,35 @@ class _RenderSwitch extends RenderToggleable {
 
   Color _thumbColor;
   Color get thumbColor => _thumbColor;
-
   void set thumbColor(Color value) {
     if (value == _thumbColor) return;
     _thumbColor = value;
     markNeedsPaint();
   }
 
+  RadialReaction _radialReaction;
+
+  EventDisposition handleEvent(sky.Event event, BoxHitTestEntry entry) {
+    if (event is sky.PointerEvent && event.type == 'pointerdown') {
+      _startRadialReaction(entry.localPosition);
+      return EventDisposition.processed;
+    }
+    return EventDisposition.ignored;
+  }
+
+  Future _startRadialReaction(Point startLocation) async {
+    if (_radialReaction != null)
+      return;
+    _radialReaction = new RadialReaction(
+      center: new Point(_kSwitchSize.width / 2.0, _kSwitchSize.height / 2.0),
+      radius: _kReactionRadius,
+      startPosition: startLocation)
+      ..addListener(markNeedsPaint);
+    await _radialReaction.forward();
```

Comment:
`forward` returns a Future that resolves when the animation is over.  This waits for the animation to be over and then removes the radial reaction.